config/config.go

```diff
@@ -361,7 +361,8 @@ type MetricsConfig struct {
 }
 
 type WorkflowRestrictions struct {
-	TemplateReferencing TemplateReferencing `json:"templateReferencing"`
+	TemplateReferencing   TemplateReferencing   `json:"templateReferencing,omitempty"`
+	DataTemplatePodPolicy DataTemplatePodPolicy `json:"dataTemplatePodPolicy,omitempty"`
```

Comment:
can I please ask you to remove this from this PR and we can follow up with >= v1.1

Reply:
I actually think this is important this is shipped with the MVP as it protects from a possible exploit. Terry's comment describes it well: https://github.com/argoproj/argo-workflows/pull/4958#pullrequestreview-592572520

Reply:
May I ask what v1.1 is? I've been seeing that a lot in recent PRs.

Reply:
v1.1 of this feature (not of Argo)

Reply:
it's slang:

* v1.0 - the MVP 
* v1.1/v1.2/v1.3 etc- an improvement to a feature we probably want follow-up soon after v1.0
* v2.0 - a complex/big improvement we want to hold off on on

Reply:
give (a) this PR will only ship with `artifactsPaths` and (b) we only provide an expression transformation - I don't think we have a security risk with this PR?

Reply:
regardless, lets narrow the scope v1.0 and follow up with a v1.1 after discussion